src/windows/exportClips.py

```diff
@@ -0,0 +1,236 @@
+        self.buttonBox.addButton(self.cancel_button, QDialogButtonBox.ActionRole)
+        self.buttonBox.addButton(QDialogButtonBox.Close).clicked.connect(lambda: self.done(0))
+        self.FilePickerArea.addWidget(self.fp)
+        self.progressExportVideo.setValue(0)
+
+    def _cancelButtonClicked(self):
+        self.exporting = False
+
+    def _updateProgressBar(self, count: int, total: int):
+        d = count - total
+        if -2 <= d and 2 >= d:
+            self.progressExportVideo.setValue(100)
+            return
+        self.progressExportVideo.setValue((count/total) * 100)
+
+    def _exportPressed(self):
+        clips = list(filter(isClip, self.file_objs))
+        files = list(filter(notClip, self.file_objs))
+        # Total number of frames
+        self._updateDialogExportStarting()
+        export_destination = self.fp.getPath()
+        total_frames, frames_written = 0, 0
+        for c in clips:
+            total_frames += framesInClip(c)
+        for f in files:
+            total_frames += int(f.data.get("video_length",0))
+        for f in files:
+            copyFileToFolder(f, export_destination)
+            frames_written+=int(f.data.get("video_length",0))
+        for c in clips:
+            extension = "mp4"
+            export_path = os.path.join(export_destination, f"{nameOfExport(c)}.{extension}")
+            if(os.path.exists(export_path)):
+                log.info("Export path exists. Skipping render")
+                frames_written += framesInClip(c)
+                self._updateProgressBar(frames_written, total_frames)
+                get_app().processEvents()
+                continue
+            w = openshot.FFmpegWriter(export_path)
+            setupWriter(c, w)
+            w.PrepareStreams()
+            w.Open()
+
+            start_time = c.data.get("start")
+            end_time = c.data.get("end")
+            fps = c.data.get("fps").get("num") / c.data.get("fps").get("den")
+            start_frame, end_frame = int(start_time * fps), int(end_time*fps)
+
+            clip_reader = openshot.Clip(c.data.get("path"))
+            clip_reader.Open()
+
+            log.info(f"Starting to write frames to {export_path}")
+            for frame in range(start_frame+1, end_frame+1):
+                w.WriteFrame(clip_reader.GetFrame(frame))
+                if frame % 5 == 0:
+                    self._updateProgressBar(frames_written, total_frames)
+                    get_app().processEvents()
+                frames_written += 1
+                if not self.exporting:
+                    break
+            clip_reader.Close()
+            w.Close()
+            log.info("Finished Exporting Clip: %s" % export_path)
+        log.info("Finished exporting")
+        self._updateProgressBar(frames_written, total_frames)
+        self._updateDialogExportFinished()
+
+    def _updateProgressBar(self, count: int, total: int):
+        if total==0:
```

Comment:
Is this function duplicated? Let's try and not duplicate it.

Reply:
Oops, I meant to move it to be next to the starting and ending export updates. The functions that disable some buttons and change the title.